dockerconverter/convert.go

```diff
@@ -0,0 +1,103 @@
+package dockerconverter
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/earthly/earthly/fileutils"
+
+	"github.com/pkg/errors"
+)
+
+func getArtifactName(s string) string {
+	split := strings.Split(s, " ")
+	n := len(split)
+	return split[n-1]
+}
+
+// DockerToEarth converts an existing Dockerfile in the current directory and writes out an Earthfile in the current directory
+// and error is returned if an Earthfile already exists.
+func DockerToEarth() error {
+	if fileutils.FileExists("Earthfile") {
+		return fmt.Errorf("Earthfile already exists; please delete it if you wish to continue")
+	}
+
+	in, err := os.Open("Dockerfile")
+	if err != nil {
+		return errors.Wrap(err, "failed to open ./Dockerfile")
+	}
+	defer in.Close()
+
+	targets := [][]string{
+		{},
+	}
+	target := 0
+	scanner := bufio.NewScanner(in)
+	for scanner.Scan() {
```

Comment:
It might be less brittle to use the dockerfile libraries themselves - like they are used here: https://github.com/moby/buildkit/blob/f4ebd6d352ac1fd5f6a691818a44b4dbbc4b09ba/frontend/dockerfile/dockerfile2llb/convert.go#L84-L105

Watch for that escape char though - Dockerfile has a way in which the user can override the escape character (it's used mostly for windows containers).

Reply:
Great suggestion, I didn't realise it was available before it was converted into LLB.